### ISSUE TITLE
Retrieve Name spaced metadata in OrgBrowser

### ISSUE THREAD

**Issue by Amit-Tripathi** [forcedotcom/salesforcedx-vscode#5651]
### Summary

Org Browser is not fetching name spaced metadata like Custom Objects, Flows, Page Layout, Custom metadata, Flexipages, Custom Settings and labels.
Org Browser is wonderful if we are working with unpackaged metadata but makes it difficult to support metadata from namespaces that are editable like objects, flows etc. 
There was a fix pushed for issue- https://github.com/forcedotcom/salesforcedx-vscode/issues/1998, but it doesn't seem to be working hence raising another one.

### Steps To Reproduce:

1. Install any Managed Package with Custom Object, Flows, Flexipage, Page Layout, and Custom metadata in an org
2. Connect it to VScode and run org browser
3. The Name spaced component does not show up as listed hence there is no ability to fetch them

### Expected result

All the components from the org with namespace should be listed in the org browser

### Actual result

Only Unmanaged metadata with no namespace is listed in the org browser

### Additional information

_Feel free to attach a screenshot_.

**Salesforce Extension Version in VS Code**: Extension Pack- v61.0.1

**Salesforce CLI Version**:  @salesforce/cli/2.45.6

**OS and version**: Windows_NT x64 10.0.22631

**VS Code version**: 1.90.2

**Most recent version of the extensions where this was working**: Not sure

**Comment by CristiCanizales** [comment]
Hi @eriksinnard, thanks for creating the issue. I was wondering if you tried refreshing the org browser (Open Command Palette > Run command `SFDX: Refresh Types`). There's a current issue with duplicated namespaces in Org Browser, but at least they show up. I'm curious on why you don't even see the metadata. For example I see it like this 
<img width="429" alt="image" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/113132642/3dd3f6f4-b626-44a3-b6cc-e62e7fa6ad87">

**Comment by eriksinnard** [comment]
@CristiCanizales I performed a refresh using the refresh button in the org browser and using the SFDX: Refresh Types command at the command palette and I still do not see any namespaced components in the org browser.

I tried this against all of the org types I have access to; scratch orgs, sandboxes, and trialforce production orgs and I see the same behavior.

To eliminate any issues with local configuration, I set up a new SFDX project in a Salesforce CodeBuilder instance and connected to a default org with several managed packages installed.  In all of these cases, I still do not see any managed components in the org browser.

![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/43754721/e08197b8-4c89-4a33-974c-283bc6a261ac)

![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/43754721/65e36d01-55ff-4161-b0d8-183445b61c21)

Please let me know if I can provide any additional detail.

Thanks!

**Comment by daphne-sfdc** [comment]
Hi @Amit-Tripathi @eriksinnard, I spoke with my team and CLI Team, and we believe the org browser is working as expected.  The reason the components in managed packages are not visible in org browser is that they shouldn't be retrieved - they're able to be used but users aren't supposed to see their code and thus we would have to figure out how to implement this limited visibility to make the components read-only.

That being said, your issue is a [very common feature request](https://github.com/forcedotcom/salesforcedx-vscode/discussions/5481), and we do agree that we haven't added adequate support for package-based development across a number of our features.  Given the recent uptick of this issue across IDEx and CLI Teams, we'll be revisiting this in our next planning session.

Thank you for your patience, and we'll keep you in the loop with any updates.

**Comment by eriksinnard** [comment]
@daphne-sfdc that is an excellent point when it comes to managed components containing code.  I used Apex Classes in my testing to correspond with @CristiCanizales example, but in the real world I wouldn't expect the org browser or the CLI to return the code of an apex class/VF page/Aura Component/LWC.  So where those components are concerned I totally agree with your analysis.

Where components are managed but can still be modified or extended by subscribers is truly the use-case we're hoping to address.  As an example, all our subscribers add custom fields to our managed objects, but because the managed objects do not show in the org browser, interacting with these components becomes more difficult.

I was expecting that this [recent PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/5403) would update the functionality to expose managed components such as custom objects, flexipages, layouts, etc in the org browser, which would be extremely useful for a lot of our internal teams and our subscribers.  With the additional context of this discussion I wonder if this PR was focused on exposing namespaced components from _unlocked_ packages but still excluding namespaced components from _managed_ packages?  I'm inclined to think that may be why @CristiCanizales and I were seeing different behavior.

**Comment by daphne-sfdc** [comment]
@eriksinnard 

> With the additional context of this discussion I wonder if this PR was focused on exposing namespaced components from unlocked packages but still excluding namespaced components from managed packages? I'm inclined to think that may be why @CristiCanizales and I were seeing different behavior.

Yes, the example that @CristiCanizales provided above is with an unlocked package - from investigating this issue we realized that namespaced components from managed and unlocked packages are treated differently, and rightfully so.

> Where components are managed but can still be modified or extended by subscribers is truly the use-case we're hoping to address. As an example, all our subscribers add custom fields to our managed objects, but because the managed objects do not show in the org browser, interacting with these components becomes more difficult.

We did not consider this use case, but now that you mention it, it makes sense why you would want managed components to be visible in the org browser and why this is such a popular feature request.  Our team will be talking about how we want to implement this during our planning.

**Comment by eriksinnard** [comment]
@daphne-sfdc That sounds great, thank you for the update!  Don't hesitate to reach out if we can provide any additional context or feedback.

**Comment by aldoforce** [comment]
While it's unbelievable that since the launch of OrgBrowser this feature is not yet available, there are two good news: 

1) a work item is loaded in Gus, which i'm sure it's already assigned to someone and a fix will be available soon.
2) the CLI provides a mechanism to do this with ease:

```
|  Retrieve a custom object called ExcitingObject that's in the SBQQ namespace:           
│  $ sf project retrieve start --metadata CustomObject:SBQQ__ExcitingObject 
│                                     
│  Retrieve all custom objects in the SBQQ namespace by using a wildcard and quotes:                           
│  $ sf project retrieve start --metadata 'CustomObject:SBQQ__*'                 
```
